core/shared/src/main/scala-3/zio/prelude/EqualVersionSpecific.scala

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2023 John A. De Goes and the ZIO Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package zio.prelude
+
+import scala.deriving.*
+import scala.compiletime.{erasedValue, summonInline}
+
+trait EqualVersionSpecific {
+
+  inline given derived[A](using mirror: Mirror.Of[A]): Equal[A] =
```

Comment:
```suggestion
  inline def derived[A](using mirror: Mirror.Of[A]): Equal[A] =
```
Otherwise it will derive for all types with no control right?

Reply:
`given` is in the documentation of this feature and using `def` creates an infinite loop in the derivation.

Reply:
That's what I've been alluding to in my comment above: https://github.com/zio/zio-prelude/pull/1079#discussion_r1106119518

I also think that people would like to control it with `derives Equal` clauses.

Reply:
I opened an issue with the Scala 3 team regarding this.

Reply:
> `def` creates an infinite loop in the derivation.

Erf that is ... weird. This code is equivalent to fully automatic derivation. With a `def` users  would need to explicitly add the `derives` clause. I had many complaints with scanamo when instances were created implicitly.

 
I read the docs, the "recursively auto-derive" is an anti pattern imo, oh well I guess time will tell.

Reply:
So also according to @bishabosha we shouldn't put the `given` on the derived method to let the derivations be driven by `derives` clauses.

https://github.com/lampepfl/dotty/issues/16916

> using `def` creates an infinite loop in the derivation

Are you able replicate it? Maybe we should report that.

I'm trying my lack with changing the official documentation https://github.com/lampepfl/dotty/pull/16919 Let's see where this leads :smile:

Reply:
> `given` is in the documentation of this feature and using `def` creates an infinite loop in the derivation.

is this with any type, or a specific example?

Reply:
The derivation for `Color`.